plugins/single_plugins/zoom.cpp

```diff
@@ -29,41 +30,45 @@ class wayfire_invert_screen : public wayfire_plugin_t
             axis = [=] (wlr_event_pointer_axis* ev)
             {
                 if (ev->orientation == WLR_AXIS_ORIENTATION_VERTICAL)
-                    update_zoom(ev->delta);
+                    update_zoom_target(ev->delta);
             };
 
             auto section = config->get_section("zoom");
             modifier = section->get_option("modifier", "<super>");
-            speed    = section->get_option("speed", "0.1");
-
             output->add_axis(modifier, &axis);
+
+            speed    = section->get_option("speed", "0.005");
+            smoothing_duration = section->get_option("smoothing_duration", "300");
+
+            duration = wf_duration(smoothing_duration, wf_animation::linear);
+            duration.start(1, 1); // so that the first value we get is correct
         }
 
-        void update_zoom(float delta)
+        void update_zoom_target(float delta)
         {
-            zoom += delta * speed->as_cached_double();
-            zoom = std::max(zoom, 1.0f);
-            zoom = std::min(zoom, 50.0f);
+            const float last_target = target_zoom;
 
-            if (zoom - 1 <= 0.01)
+            target_zoom -= target_zoom * delta * speed->as_cached_double();
+            target_zoom = std::max(target_zoom, 1.0f);
+            target_zoom = std::min(target_zoom, 50.0f);
+
+            if (!hook_set)
```

Comment:
weren't you going to put that inside the next if()?

Reply:
Good point.